store/parameter_store.go

```diff
@@ -494,6 +495,34 @@ func NewParameterStore(maxResultsToQuery int64, env string, assume bool) *Parame
 	}
 }
 
+// NewParameterStoreWithAssumedRole creates a parameter store using an assumed role
+func NewParameterStoreWithAssumedRole(ctx context.Context, limit int, env string, roleArn string) (SecretStore, error) {
+	// Load default config (uses Main--EngineerNew credentials)
+	baseCfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Assume the IdentityEngineer role
```

Comment:
nit: This is a public library. Although heavily tailored to us, leaking internal abstractions about intended roles shouldn't be done in comments here.

```suggestion
	// Load default config
	baseCfg, err := config.LoadDefaultConfig(ctx)
	if err != nil {
		return nil, err
	}

	// Assume a particular role for secrets handling
```